File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
N = int(input())
first = N

for i in range(N):
    word = input()
    for j in range(len(word) - 1):
        if word[j] == word[j + 1]:
            pass
        elif word[j] in word[j + 1:]:
            first -= 1
            break

print(first)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int first = N;
        scanner.nextLine();

        for (int i = 0; i < N; i++) {
            String word = scanner.nextLine();
            for (int j = 0; j < word.length() - 1; j++) {
                if (word.charAt(j) == word.charAt(j + 1)) {
                    continue;
                }
                else if (word.substring(j + 1).contains(String.valueOf(word.charAt(j)))) {
                    first -= 1;
                    break;
                }
            }
        }
        System.out.println(first);
    }
}
